### ISSUE TITLE
Migration from 2.0.0 broken in 2.2.6

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5570]
2.2.6 generates a bunch of different errors during migration from a 2.0.0 database folder (an empty database is enough, no tables are needed to reproduce this):
```C++
Running rethinkdb 2.2.6 (debug) (CLANG 3.5.0 (tags/RELEASE_350/final))...
Running on Linux 3.13.0-79-generic x86_64
Loading data from directory /tmp/daniel/rethinkdb_data
Migrating cluster metadata to v2.2
Version: rethinkdb 2.2.6 (debug) (CLANG 3.5.0 (tags/RELEASE_350/final))
error: Error in src/clustering/administration/persist/migrate/migrate_v1_16.cc at line 136:
error: Guarantee failed: [res == archive_result_t::SUCCESS] Deserialization of T (template code) failed with error archive_result_t::SOCK_EOF.
error: Backtrace:
error: Mon Mar 28 16:42:55 2016
       
       1 [0x1fe628b]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2 [0x1ff1900]: backtrace_t::backtrace_t() at backtrace.cc:204 (discriminator 1)
       3 [0x1ff2cbc]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:325
       4 [0x1ff1587]: format_backtrace(bool) at backtrace.cc:199
       5 [0x201546e]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6 [0x17c767a]: /home/daniel/rethinkdb/build/debug_clang/rethinkdb() [0x17c767a] at 0x17c767a ()
       7 [0x17c72d1]: migrate_cluster_metadata_to_v2_2(io_backender_t*, base_path_t const&, buf_parent_t, void const*, metadata_file_t::write_txn_t*, signal_t*) at migrate_v1_16.cc:564 (discriminator 1)
       8 [0x1831c24]: metadata_file_t::metadata_file_t(io_backender_t*, base_path_t const&, perfmon_collection_t*, signal_t*) at file.cc:328
       9 [0x16d0440]: run_rethinkdb_serve(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, server_id_t const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) at command_line.cc:766 (discriminator 1)
       10 [0x16d0c42]: run_rethinkdb_porcelain(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*) at command_line.cc:834
       11 [0x16e0c4a]: void std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)>::__call<void, , 0ul, 1ul, 2ul, 3ul, 4ul, 5ul, 6ul, 7ul, 8ul, 9ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul, 5ul, 6ul, 7ul, 8ul, 9ul>) at functional:1295
       12 [0x16e09e6]: void std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)>::operator()<, void>() at functional:1353
       13 [0x16e012d]: std::_Function_handler<void (), std::_Bind<void (*(base_path_t, name_string_t, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> >, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, bool, serve_info_t*, directory_lock_t*, bool*))(base_path_t const&, name_string_t const&, std::set<name_string_t, std::less<name_string_t>, std::allocator<name_string_t> > const&, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, bool, serve_info_t*, directory_lock_t*, bool*)> >::_M_invoke(std::_Any_data const&) at functional:2073
       14 [0x143cb5e]: std::function<void ()>::operator()() const at functional:2471
       15 [0x147a3a1]: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       16 [0x147aa2a]: void std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()<std::function<void ()>&, void>(starter_t*, std::function<void ()>&) const at functional:601
       17 [0x147a990]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1295
       18 [0x147a916]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::operator()<, void>() at functional:1353
       19 [0x147a5ad]: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:2073
       20 [0x143cb5e]: std::function<void ()>::operator()() const at functional:2471
       21 [0x147a299]: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
       22 [0x1488b86]: callable_action_wrapper_t::run() at runtime_utils.cc:43
       23 [0x147b81d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
       24 [0x147a09d]: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&) at coroutines.hpp:62
       25 [0x147a039]: starter_t::on_thread_switch() at runtime.cc:57
       26 [0x1487e36]: linux_message_hub_t::on_event(int) at message_hub.cc:154
       27 [0x1472c07]: epoll_event_queue_t::run() at epoll.cc:115
       28 [0x14842a3]: linux_thread_pool_t::start_thread(void*) at thread_pool.cc:164
       29 [0x7f733145d182]: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8182) [0x7f733145d182] at 0x7f733145d182 (/lib/x86_64-linux-gnu/libpthread.so.0)
       30 [0x7f733118a47d]: clone+0x6d at 0x7f733118a47d (/lib/x86_64-linux-gnu/libc.so.6)
error: Exiting.
Trace/breakpoint trap (core dumped)
```

It's still working with 2.2.5.

**Comment by frankleng** [comment]
confirmed upgrading to 2.1.x or 2.2.5 as workaround

**Comment by danielmewes** [comment]
A fix is up in review 3551, commit https://github.com/rethinkdb/rethinkdb/commit/17364f666c4a80fe2380e8d43aba232674ed377a

**Comment by danielmewes** [comment]
Fixed in v2.2.x 17364f666c4a80fe2380e8d43aba232674ed377a and in next 573c853.
